extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php

```diff
@@ -43,10 +48,63 @@ public function index() {
 			'recentlyJoined' => $this->sendRequest( 'CommunityPageSpecialController', 'getRecentlyJoinedData' )
 				->getData(),
 			'recentActivityModule' => $this->getRecentActivityData(),
+			'popularPages' => $this->getInsightModule( 'popularpages' )
 		] );
 	}
 
 	/**
+	 * @param string $type type of module we want to build.
+	 * @return array Insight Module
+	 */
+	private function getInsightModule( $type ) {
+		$insightPages['pages'] = $this->insightsService->getInsightPages(
+			$type,
+			self::INSIGHT_MODULE_ITEMS,
+			self::INSIGHT_MODULE_SORT_TYPE
+		);
+
+		$insightPages['title'] = $this->msg( 'communitypage-popularpages-title' )->text();
+		$insightPages['description'] =  $this->msg( 'communitypage-popularpages-description' )->text();
+		$insightPages['edittext'] = $this->msg( 'communitypage-page-list-edit' )->text();
+		$insightPages['fulllist'] = $this->msg( 'communitypage-full-list' )->text();
+		$insightPages['fulllistlink'] = SpecialPage::getTitleFor( 'Insights', $type )->getLocalURL();
+
+		return $this->addLastRevision( $insightPages );
+	}
+
+	/**
+	 * @param array $insightsPages
+	 * @return array Prepare message about who and when last edited given article
+	 * @throws MWException
+	 */
+	private function addLastRevision( $insightsPages ) {
+		foreach ( $insightsPages['pages'] as $key => $insight ) {
+			$timestamp = wfTimestamp( TS_UNIX, $insight['metadata']['lastRevision']['timestamp'] );
+			$insightsPages['pages'][$key]['lastRevision'] = $this->msg( 'communitypage-lastrevision' )->rawParams(
+				Html::element(
+					'a',
+					['href' => $insight['metadata']['lastRevision']['userpage']],
+					$insight['metadata']['lastRevision']['username']
+				),
+				$this->getLang()->userDate( $timestamp, $this->getUser() )
+			)->escaped();
+			$insightsPages['pages'][$key]['pageviews'] = $this->msg(
+				'communitypage-noofviews',
+				$insight['metadata']['pv7']
+			)->text();
+			$insightsPages['pages'][$key]['editlink'] = $this->getEditUrl( $insight['link']['url'] );
+		}
+		return $insightsPages;
+	}
+
+	private function getEditUrl( $articleUrl ) {
+		if( EditorPreference::isVisualEditorPrimary() && $this->getUser()->isLoggedIn() ) {
```

Comment:
Why are we only linking to VE if the user is logged in? Anonymous users should get VE as well, right?

Also, minor: there should be a space between `if` and the opening parenthesis.